controllers/operandrequest/operandrequest_controller_test.go

```diff
@@ -160,27 +180,78 @@ var _ = Describe("OperandRegistry controller", func() {
+			// 	Expect(k8sClient.Update(ctx, requestInstance2)).Should(Succeed())
+			// 	etcdCluster := &v1beta2.EtcdCluster{}
+			// 	err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: "example", Namespace: operatorNamespaceName}, etcdCluster)
+			// 	return err != nil && errors.IsNotFound(err)
+			// }, testutil.Timeout, testutil.Interval).Should(BeTrue())
+
+			// By("Checking the etcd operator has been deleted")
+			// Eventually(func() bool {
+			// 	etcdCSV := &olmv1alpha1.ClusterServiceVersion{}
+			// 	err := k8sClient.Get(ctx, types.NamespacedName{Name: "etcd-csv.v0.0.1", Namespace: operatorNamespaceName}, etcdCSV)
+			// 	return err != nil && errors.IsNotFound(err)
+			// }, testutil.Timeout, testutil.Interval).Should(BeTrue())
+
+			By("Deleting the first OperandRequest")
+			Expect(k8sClient.Delete(ctx, request1)).Should(Succeed())
+
+			By("Checking jenkins operator has not been deleted")
+			Eventually(func() error {
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
+				return err
+			}, testutil.Timeout, testutil.Interval).Should(Succeed())
+
+			By("Checking etcd operator has not been deleted")
+			Eventually(func() error {
+				etcdCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "etcd-csv.v0.0.1", Namespace: operatorNamespaceName}, etcdCSV)
+				return err
+			}, testutil.Timeout, testutil.Interval).Should(Succeed())
+
+			By("Checking jenkins operator has been deleted")
 			Eventually(func() bool {
-				requestInstance := &operatorv1alpha1.OperandRequest{}
-				Expect(k8sClient.Get(ctx, requestKey, requestInstance)).Should(Succeed())
-				requestInstance.Spec.Requests[0].Operands = requestInstance.Spec.Requests[0].Operands[1:]
-				Expect(k8sClient.Update(ctx, requestInstance)).Should(Succeed())
-				etcdCluster := &v1beta2.EtcdCluster{}
-				err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: "example", Namespace: operatorNamespaceName}, etcdCluster)
+				jenkinsCSV := &olmv1alpha1.ClusterServiceVersion{}
+				err := k8sClient.Get(ctx, types.NamespacedName{Name: "jenkins-csv.v0.0.1", Namespace: operatorNamespaceName}, jenkinsCSV)
 				return err != nil && errors.IsNotFound(err)
 			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
 
-			By("Deleting the OperandRequest")
-			Expect(k8sClient.Delete(ctx, request)).Should(Succeed())
-
-			By("Checking operators have been deleted")
+			By("Checking etcd operator has been deleted")
 			Eventually(func() bool {
 				etcdCSV := &olmv1alpha1.ClusterServiceVersion{}
 				err := k8sClient.Get(ctx, types.NamespacedName{Name: "etcd-csv.v0.0.1", Namespace: operatorNamespaceName}, etcdCSV)
 				return err != nil && errors.IsNotFound(err)
 			}, testutil.Timeout, testutil.Interval).Should(BeTrue())
```

Comment:
If I did not disable `etcd` operator before, so both `OperandRequest` have `etcd` and `jejenkins`.

Then directly start to delete the first `OperandRequest`, this time `jenkins` operator exists and line 241 failed because it should be `Should(BeFalse())`

So it may not the problem of `jenkins` operator.

Here is the failed test link
http://prow.purple-chesterfield.com/log?job=test-operand-deployment-lifecycle-manager-amd64&id=1374232080648507394